src/hpr_config_worker.erl

```diff
@@ -87,59 +86,40 @@ init(#{host := Host, port := Port} = Args) ->
 
 handle_continue(?CONNECT, #state{host = Host, port = Port, conn_backoff = Backoff0} = State) ->
     lager:info("connecting"),
-    case grpc_client:connect(tcp, Host, Port) of
+    case grpcbox_client:connect(?CHANNEL_NAME, [{http, Host, Port, []}], #{}) of
         {ok, Connection} ->
             lager:info("connected"),
             {_, Backoff1} = backoff:succeed(Backoff0),
             {noreply, State#state{connection = Connection, conn_backoff = Backoff1},
                 {continue, ?INIT_STREAM}};
+        {error, {already_started, Connection}} ->
+            lager:info("connection already started"),
+            {noreply, State#state{connection = Connection}, {continue, ?INIT_STREAM}};
         {error, _E} ->
             {Delay, Backoff1} = backoff:fail(Backoff0),
             lager:error("failed to connect ~pm sleeping ~wms", [_E, Delay]),
             timer:sleep(Delay),
             {noreply, State#state{conn_backoff = Backoff1}, {continue, ?CONNECT}}
     end;
-handle_continue(
-    ?INIT_STREAM,
-    #state{
-        connection = Connection
-    } = State
-) ->
-    {ok, Stream} = grpc_client:new_stream(
-        Connection, 'helium.config.route', stream, client_config_pb
-    ),
-    %% Sending Route Stream Request
+handle_continue(?INIT_STREAM, #state{conn_backoff = Backoff0} = State) ->
     {PubKey, SigFun} = persistent_term:get(?HPR_KEY),
     PubKeyBin = libp2p_crypto:pubkey_to_bin(PubKey),
     RouteStreamReq = hpr_route_stream_req:new(PubKeyBin, erlang:system_time(millisecond)),
     SignedRouteStreamReq = hpr_route_stream_req:sign(RouteStreamReq, SigFun),
-    ok = grpc_client:send_last(Stream, hpr_route_stream_req:to_map(SignedRouteStreamReq)),
-    lager:info("stream initialized"),
-    {noreply, State#state{stream = Stream}, {continue, ?RCV_CFG_UPDATE}};
-handle_continue(
-    ?RCV_CFG_UPDATE,
-    #state{
-        connection = Connection, stream = Stream, file_backup_path = Path, conn_backoff = Backoff0
-    } = State
-) ->
-    case grpc_client:rcv(Stream, ?RCV_TIMEOUT) of
-        {headers, _Headers} ->
-            {noreply, State, {continue, ?RCV_CFG_UPDATE}};
-        {data, RouteStreamRes} ->
-            lager:info("got router update"),
-            ok = process_route_stream_res(hpr_route_stream_res:from_map(RouteStreamRes), Path),
-            {noreply, State, {continue, ?RCV_CFG_UPDATE}};
-        eof ->
-            lager:warning("got eof"),
-            _ = catch grpc_client:stop_connection(Connection),
-            {noreply, State, {continue, ?CONNECT}};
-        {error, timeout} ->
-            lager:debug("rcv timeout"),
-            {_, Backoff1} = backoff:succeed(Backoff0),
-            {noreply, State#state{conn_backoff = Backoff1}, {continue, ?RCV_CFG_UPDATE}};
-        {error, E} ->
-            lager:error("failed to rcv ~p", [E]),
-            {stop, {error, E}}
+
+    %% NOTE: Client is still expecting maps.
```

Comment:
I thought GRPC box expected records?

Reply:
For posterity:

We use [grpcbox_plugin](https://github.com/novalabsxyz/grpcbox_plugin).
Grpcbox can be configured to use maps _or_ records.

`grpc_client` _only_ supports maps.
So we configured our client modules to be generated using maps.

For the changover from `grpc_client` to `grpcbox_client`, only the client handlers are being swapped out.